### PR TITLE
Co-Pilot: Stream Response and Vector Chunk Search Draft

### PR DESCRIPTION
**Pull request by Vivek-Lahole**
# Description 📣

## Type ✨

- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://github.com/OcularEngineering/ocular/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: packages/ocular/src/services/chat.ts [hunk 1]
```diff
@@ -66,9 +66,50 @@ class ChatService extends TransactionBaseService {
     return chat
   }
 
-  async chat(chatId: string, message:string, context: ChatContext): Promise<ChatResponse> {
-    return await this.atomicPhase_(
-      async (transactionManager: EntityManager) => {
+  // async chat(chatId: string, message:string, context: ChatContext): Promise<ChatResponse> {
+  //   return await this.atomicPhase_(
+  //     async (transactionManager: EntityManager) => {
```
Comment (louismurerwa) [inline packages/ocular/src/services/chat.ts:71]: Do we still need this uncommented code?

File: packages/ocular/src/approaches/chat-retrieve-read-retrieve.ts [hunk 2]
```diff
@@ -79,68 +81,65 @@ export default class ChatReadRetrieveRead implements IChatApproach {
...
-      //     },
-      //   },
-      // ],
-      message: {
-        role: "assistant",
-        content: chatCompletion,
-      },
-      data_points: hits,
-    };
+    // const { completionRequest, thoughts } = await this.baseRun(
```
Comment (louismurerwa) [inline packages/ocular/src/approaches/chat-retrieve-read-retrieve.ts:84]: @Vivek-Lahole do we still need this uncommented block?

File: packages/ocular-ui/components/chat/chat-helpers/index.tsx [hunk 3]
```diff
@@ -60,18 +49,59 @@ export const fetchChatResponse = async (
...
+              } else {
+                const lastMessage = updatedMessages[updatedMessages.length - 1];
+                lastMessage.message.content = chunk.choices[0].delta.content;
+                lastMessage.message.updated_at = new Date();
+              }
+              return updatedMessages;
+            });
+          }
+          
+          // setChatMessages(prevChatMessages => [...prevChatMessages,
```
Comment (louismurerwa) [inline packages/ocular-ui/components/chat/chat-helpers/index.tsx:85]: @Vivek-Lahole do we still need this too?

File: packages/ocular/src/approaches/chat-retrieve-read-retrieve.ts (hunk 2 above)
Comment (Vivek-Lahole) [reply packages/ocular/src/approaches/chat-retrieve-read-retrieve.ts:84]: yes , need to clean that up for a non-streaming response

File: packages/ocular/src/services/chat.ts (hunk 1 above)
Comment (Vivek-Lahole) [reply packages/ocular/src/services/chat.ts:71]: no we dont need it , just a heads up we need to make all transactrion for the method  , yet to be implemented

File: packages/ocular-ui/components/chat/chat-helpers/index.tsx (hunk 3 above)
Comment (Vivek-Lahole) [reply packages/ocular-ui/components/chat/chat-helpers/index.tsx:85]: no dont need that

**Review by louismurerwa — APPROVED** [review]

**Pull request merged**